### ISSUE TITLE
Amplify deployment Failed with `Build Artifacts not found` error.

### ISSUE THREAD

**Issue by hoseinrafiei** [aws-amplify/amplify-hosting#3503]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-hosting/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-hosting/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### App Id

d1eqfjmspof9g8

### AWS Region

us-west-2

### Amplify Hosting feature

Deployments

### Describe the bug

I'm facing following error after a sucessfull build.

```
[INFO]: Beginning deployment for application [APP_ID], branch:[BRANCH], buildId [ID]
[INFO]: Build Artifacts not found.
[ERROR]: {"code":"MissingBuildArtifacts"}
```
I checked and tried #3249 and #3282 issues. Since my application is a React project without next.js framework, I decided to create a new issue.

### Expected behavior

The code should get deployed because there is no error in provision and build processes.

### Reproduction steps

1. Push a new change into the connected branch or redeploy a version.

### Build Settings

```yaml
version: 1
applications:
  - appRoot: apps/user
    frontend:
      phases:
        preBuild:
          commands:
            - nvm install 16.13.0
            - nvm use 16.13.0
            - yarn install
            - cd ../..
            - yarn install
        build:
          commands:
            - nvm install 16.13.0
            - nvm use 16.13.0
            - pwd
            - yarn build user
        postBuild:
          commands:
            - pwd
      artifacts:
        baseDirectory: ../../dist/apps/user
        files:
          - '**/*'
      cache:
        paths:
          - node_modules/**/*
  - appRoot: apps/instructor
    frontend:
      phases:
        build:
          commands:
            - nvm install 16.13.0
            - nvm use 16.13.0
            - cd ../..
            - yarn install
            - yarn nx export instructor
      artifacts:
        baseDirectory: ../../dist/apps/instructor/exported
        files:
          - '**/*'
      cache:
        paths:
          - node_modules/**/*
```

### Log output

<details>

```
# Put your logs below this line
                                 # Starting phase: preBuild
                                 # Executing command: nvm install 16.13.0
2023-05-29T22:25:30.182Z [INFO]: Downloading and installing node v16.13.0...
2023-05-29T22:25:30.303Z [WARNING]: Downloading https://nodejs.org/dist/v16.13.0/node-v16.13.0-linux-x64.tar.gz...
2023-05-29T22:25:30.420Z [WARNING]: #
2023-05-29T22:25:30.420Z [WARNING]: 1.6%
2023-05-29T22:25:30.519Z [WARNING]: ##########################
2023-05-29T22:25:30.521Z [WARNING]: #######                                         46.7%
2023-05-29T22:25:30.632Z [WARNING]: ############################
2023-05-29T22:25:30.632Z [WARNING]: ###########################################   99.2%
2023-05-29T22:25:30.633Z [WARNING]: ############
2023-05-29T22:25:30.633Z [WARNING]: ############################################################ 100.0%
2023-05-29T22:25:30.648Z [WARNING]: Computing checksum with sha256sum
2023-05-29T22:25:30.726Z [WARNING]: Checksums matched!
2023-05-29T22:25:31.840Z [INFO]: Now using node v16.13.0 (npm v8.1.0)
2023-05-29T22:25:31.869Z [INFO]: Installing default global packages from /root/.nvm/default-packages...
2023-05-29T22:25:31.869Z [INFO]: npm install -g --quiet yarn@1.22.0
2023-05-29T22:25:33.061Z [INFO]: added 1 package, and audited 2 packages in 1s
2023-05-29T22:25:33.063Z [INFO]: found 0 vulnerabilities
2023-05-29T22:25:33.067Z [INFO]: # Executing command: nvm use 16.13.0
2023-05-29T22:25:33.359Z [INFO]: Now using node v16.13.0 (npm v8.1.0)
2023-05-29T22:25:33.359Z [INFO]: # Executing command: yarn install
2023-05-29T22:25:33.651Z [INFO]: yarn install v1.22.0
2023-05-29T22:25:33.737Z [INFO]: [1/4] Resolving packages...
2023-05-29T22:25:34.307Z [INFO]: [2/4] Fetching packages...
2023-05-29T22:26:03.436Z [INFO]: info fsevents@2.3.2: The platform "linux" is incompatible with this module.
2023-05-29T22:26:03.438Z [INFO]: info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.
2023-05-29T22:26:03.446Z [INFO]: info fsevents@1.2.13: The platform "linux" is incompatible with this module.
2023-05-29T22:26:03.447Z [INFO]: info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
2023-05-29T22:26:03.455Z [INFO]: [3/4] Linking dependencies...
2023-05-29T22:26:03.457Z [WARNING]: warning "@emotion/react > @emotion/babel-plugin@11.10.0" has unmet peer dependency "@babel/core@^7.0.0".
2023-05-29T22:26:03.458Z [WARNING]: warning "@emotion/react > @emotion/babel-plugin > @babel/plugin-syntax-jsx@7.18.6" has unmet peer dependency "@babel/core@^7.0.0-0".
2023-05-29T22:26:03.460Z [WARNING]: warning " > @testing-library/user-event@12.8.3" has unmet peer dependency "@testing-library/dom@>=7.21.4".
2023-05-29T22:26:03.461Z [WARNING]: warning "aws-amplify > @aws-amplify/analytics > @aws-sdk/client-firehose > @aws-sdk/middleware-retry > react-native-get-random-values@1.8.0" has unmet peer dependency "react-native@>=0.56".
2023-05-29T22:26:03.461Z [WARNING]: warning " > bootstrap@5.2.0" has unmet peer dependency "@popperjs/core@^2.11.5".
2023-05-29T22:26:03.475Z [WARNING]: warning " > styled-components@5.3.5" has unmet peer dependency "react-is@>= 16.8.0".
2023-05-29T22:26:40.718Z [INFO]: [4/4] Building fresh packages...
2023-05-29T22:26:50.384Z [INFO]: Done in 76.74s.
2023-05-29T22:26:50.433Z [INFO]: # Executing command: cd ../..
2023-05-29T22:26:50.491Z [INFO]: # Executing command: yarn install
2023-05-29T22:26:50.614Z [INFO]: yarn install v1.22.0
2023-05-29T22:26:50.706Z [INFO]: [1/4] Resolving packages...
2023-05-29T22:26:51.262Z [INFO]: success Already up-to-date.
2023-05-29T22:26:51.265Z [INFO]: Done in 0.66s.
2023-05-29T22:26:51.273Z [INFO]: # Completed phase: preBuild
2023-05-29T22:26:51.273Z [INFO]: # Starting phase: build
                                 # Executing command: nvm install 16.13.0
2023-05-29T22:26:51.666Z [WARNING]: v16.13.0 is already installed.
2023-05-29T22:26:51.961Z [INFO]: Now using node v16.13.0 (npm v8.1.0)
2023-05-29T22:26:51.961Z [INFO]: Installing default global packages from /root/.nvm/default-packages...
                                 npm install -g --quiet yarn@1.22.0
2023-05-29T22:26:52.397Z [INFO]: changed 1 package, and audited 2 packages in 330ms
2023-05-29T22:26:52.398Z [INFO]: found 0 vulnerabilities
2023-05-29T22:26:52.407Z [INFO]: # Executing command: nvm use 16.13.0
2023-05-29T22:26:52.704Z [INFO]: Now using node v16.13.0 (npm v8.1.0)
2023-05-29T22:26:52.704Z [INFO]: # Executing command: pwd
                                 /codebuild/output/src959274480/src/frontry
2023-05-29T22:26:52.704Z [INFO]: # Executing command: yarn build user
2023-05-29T22:26:52.826Z [INFO]: yarn run v1.22.0
2023-05-29T22:26:52.848Z [INFO]: $ nx build user
2023-05-29T22:26:55.372Z [INFO]: 
                                 >  NX   Successfully ran target build for project user
                                 Nx read the output from the cache instead of running the command for 1 out of 1 tasks.
2023-05-29T22:26:55.606Z [INFO]: Nx Cloud made it possible to reuse user:
2023-05-29T22:26:55.615Z [INFO]: Done in 2.79s.
2023-05-29T22:26:55.619Z [INFO]: # Completed phase: build
                                 # Starting phase: postBuild
2023-05-29T22:26:55.620Z [INFO]: # Executing command: pwd
                                 /codebuild/output/src959274480/src/frontry
                                 # Completed phase: postBuild
2023-05-29T22:26:55.622Z [INFO]: ## Build completed successfully
2023-05-29T22:26:55.623Z [INFO]: # Starting caching...
2023-05-29T22:26:55.633Z [INFO]: # Creating cache artifact...
2023-05-29T22:27:15.085Z [INFO]: # Created cache artifact
2023-05-29T22:27:15.200Z [INFO]: # Uploading cache artifact...
2023-05-29T22:27:18.746Z [INFO]: # Uploaded cache artifact
2023-05-29T22:27:18.815Z [INFO]: # Caching completed
2023-05-29T22:27:18.970Z [INFO]: # Last deployed commit: 10d515cfc3b1845cfad715335356a4f2c887ad28
2023-05-29T22:27:19.013Z [INFO]: # No custom headers found.
2023-05-29T22:27:19.059Z [INFO]: # Starting build artifact upload process...
2023-05-29T22:27:19.683Z [INFO]: # Uploading build artifact '__artifacts.zip'...
2023-05-29T22:27:19.706Z [INFO]: # Uploading build artifact '__artifactsHash.zip'...
2023-05-29T22:27:20.006Z [INFO]: # Build artifact upload completed
2023-05-29T22:27:20.007Z [INFO]: # Starting environment caching...
2023-05-29T22:27:20.007Z [INFO]: # Uploading environment cache artifact...
2023-05-29T22:27:20.068Z [INFO]: # Uploaded environment cache artifact
2023-05-29T22:27:20.068Z [INFO]: # Environment caching completed
Terminating logging...

```

</details>

### Additional information

Build is successful. I downloaded the build and served it in local.
This happened suddenly without any meaningful change to the app settings, code, etc. The deployment process was working fine 5 days ago but the problem started Today.

Even I cannot redeploy the versions that was deployed successfully before.

**Comment by nadetastic** [comment]
Hi @hoseinrafiei thank you for opening this issue. I took a look at the app id and noticed that your app is getting detected as an SSR app but from you comment you mentioned that this is not an SSR/Next app just plain react.

I also see that this looks like its configured from a mono repo. I'm discussing this with the team and will follow up soon. Let me know if you have any other question in the meantime - thanks!

**Comment by hoseinrafiei** [comment]
FYI

## Dependencies
<details>

```
"dependencies": {
    "@reduxjs/toolkit": "^1.8.4",
    "aws-amplify": "^4.3.31",
    "core-js": "^3.6.5",
    "date-fns": "^2.29.2",
    "next": "12.2.3",
    "react": "18.2.0",
    "react-datepicker": "^4.8.0",
    "react-dom": "18.2.0",
    "react-is": "18.2.0",
    "react-redux": "^8.0.2",
    "regenerator-runtime": "0.13.7",
    "styled-components": "5.3.5",
    "tslib": "^2.3.0"
  },
  "devDependencies": {
    "@craco/craco": "^6.4.5",
    "@nrwl/cli": "14.5.5",
    "@nrwl/cypress": "14.5.5",
    "@nrwl/eslint-plugin-nx": "14.5.5",
    "@nrwl/jest": "14.5.5",
    "@nrwl/linter": "14.5.5",
    "@nrwl/next": "^14.5.5",
    "@nrwl/nx-cloud": "latest",
    "@nrwl/react": "14.5.5",
    "@nrwl/web": "14.5.5",
    "@nrwl/workspace": "14.5.5",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "13.3.0",
    "@types/jest": "27.4.1",
    "@types/node": "16.11.7",
    "@types/react": "18.0.17",
    "@types/react-dom": "18.0.6",
    "@types/react-is": "17.0.3",
    "@types/styled-components": "5.1.25",
    "@typescript-eslint/eslint-plugin": "^5.29.0",
    "@typescript-eslint/parser": "^5.29.0",
    "babel-jest": "27.5.1",
    "babel-plugin-styled-components": "1.10.7",
    "cypress": "^10.2.0",
    "eslint": "~8.15.0",
    "eslint-config-next": "12.2.3",
    "eslint-config-prettier": "8.1.0",
    "eslint-config-react-app": "^7.0.1",
    "eslint-plugin-cypress": "^2.10.3",
    "eslint-plugin-import": "2.26.0",
    "eslint-plugin-jsx-a11y": "6.6.1",
    "eslint-plugin-react": "7.30.1",
    "eslint-plugin-react-hooks": "4.6.0",
    "jest": "27.5.1",
    "jest-watch-typeahead": "^2.0.0",
    "nx": "14.5.5",
    "prettier": "^2.6.2",
    "react-scripts": "^5.0.1",
    "react-test-renderer": "18.2.0",
    "ts-jest": "27.1.4",
    "ts-node": "10.9.1",
    "typescript": "~4.7.2",
    "web-vitals": "^2.1.4"
  }
```

</details>

**Comment by nadetastic** [comment]
Thanks for the additional info @hoseinrafiei - It looks like `staging-2`  was detected as `Next.js - SSR` which caused the platform to update to the `WEB_DYNAMIC` (SSR)

Did you add the Next dependency in that branch? if so it may have triggered the platform to change which is an app level setting. To unblock the build you can try change the platform back to WEB  via the [UpdateApp](https://docs.aws.amazon.com/amplify/latest/APIReference/API_UpdateApp.html) api.

An example of how to do this with the AWS CLI:
```
aws amplify update-app --app-id <APP_ID> --platform WEB --region <REGION>
```

**Comment by error-try-again** [comment]
I had the same issue as @hoseinrafiei and struggled with it all day, and your CLI solution allowed me to solve it. Thank you!

> Thanks for the additional info @hoseinrafiei - It looks like `staging-2` was detected as `Next.js - SSR` which caused the platform to update to the `WEB_DYNAMIC` (SSR)
> 
> Did you add the Next dependency in that branch? if so it may have triggered the platform to change which is an app level setting. To unblock the build you can try change the platform back to WEB via the [UpdateApp](https://docs.aws.amazon.com/amplify/latest/APIReference/API_UpdateApp.html) api.
> 
> An example of how to do this with the AWS CLI:
> 
> ```
> aws amplify update-app --app-id <APP_ID> --platform WEB --region <REGION>
> ```

**Comment by hoseinrafiei** [comment]
@nadetastic I fixed the issue for two of my apps using the command you sent but I'm still having issue with the rest of apps. They keep changing to `Web dynamic` and `Next.js - SSR` upon receiving an update from Github.

Is there anyway that I can set the platform and framework using command line and disable the amplify auto platform change?

**Comment by hoseinrafiei** [comment]
@nadetastic Would you please update me?

**Comment by nadetastic** [comment]
@hoseinrafiei out of curiosity, you mentioned that the two apps you have are not next apps - do you maybe have a third app that you aren't deploying on Amplify that is using NextJS, or what is the reason to have the NextJS dependency?

**Comment by hloriii** [comment]
@hoseinrafiei Amplify auto-detects the framework and platform for your application based on the dependencies in your `package.json` and there isn't a way to disable this behavior. If you're still running into this issue, please provide the app IDs for the rest of the applications that you're observing this behavior on.
